docs/src/snap/howto/install/install-offline.md

```diff
@@ -0,0 +1,337 @@
+```{note}
+With updates to the snap the base core is subject to change in the future.
+```
+
+### Network Requirements
+
+Air-gap deployments are typically associated with a number of constraints and
+restrictions when it comes to the networking connectivity of the machines.
+Below we discuss the requirements that the deployment needs to fulfill.
+
+#### Cluster node communication
+
+<!-- TODO: Add Services and Ports Doc -->
+
+Ensure that all cluster nodes are reachable from each other.
+
+<!-- Refer to [Services and ports][svc-ports] used for a list of all network
+ports used by Canonical Kubernetes.  -->
+
+#### Default Gateway
+
+In cases where the air gap environment does not have a default gateway,
+add a dummy default route on interface eth0 using the following command:
+
+```bash
+ip route add default dev eth0
+```
+
+```{note} 
+Confirm the name of the default network interface used for pod-to-pod
+communication by running "ip a".
+```
+
+```{note} 
+The dummy gateway will only be used by the Kubernetes services to 
+know which interface to use, actual connectivity to the internet is not 
+required. Ensure that the dummy gateway rule survives a node reboot.
+```
+
+#### Ensure proxy access
+
+This section is only relevant if access to upstream image registries
+(e.g. docker.io, quay.io, rocks.canonical.com, etc.)
+is only allowed through an HTTP proxy (e.g. [squid][squid]).
+
+Ensure that all nodes can use the proxy to access the image registry.
+For example, if using `http://squid.internal:3128` to access docker.io,
+an easy way to test connectivity is:
+
+```
+export https_proxy=http://squid.internal:3128
+curl -v https://registry-1.docker.io/v2
+```
+
+### Images
+
+All workloads in a Kubernetes cluster are running as an OCI image.
+Kubernetes needs to be able to fetch these images and load them
+into the container runtime, in order to run workloads.
+For a Canonical Kubernetes deployment, you will need to fetch the images
+used by its features (network, dns, etc) as well as any images that are
+needed to run your workloads.
+
+The following options are presented in the order of
+increasing complexity of implementation.
+You may also find it helpful to combine these options for your scenario.
+
+If you already have the `k8s` snap installed,
+you can list the images in use like this:
+
+```bash
+ubuntu@demo:~$ sudo k8s list-images
+ghcr.io/canonical/cilium-operator-generic:1.15.2-ck2
+ghcr.io/canonical/cilium:1.15.2-ck2
+ghcr.io/canonical/coredns:1.11.1-ck4
+ghcr.io/canonical/k8s-snap/pause:3.10
+ghcr.io/canonical/k8s-snap/sig-storage/csi-node-driver-registrar:v2.10.1
+ghcr.io/canonical/k8s-snap/sig-storage/csi-provisioner:v5.0.1
+ghcr.io/canonical/k8s-snap/sig-storage/csi-resizer:v1.11.1
+ghcr.io/canonical/k8s-snap/sig-storage/csi-snapshotter:v8.0.1
+ghcr.io/canonical/metrics-server:0.7.0-ck0
+ghcr.io/canonical/rawfile-localpv:0.8.0-ck5
+```
+
+A list of images can also be found in the downloaded k8s snap for the
+`images.txt` file.
```

Comment:
The list of images can also by looking at `images.txt`, if you unsquash the downloaded k8s.snap